mp4/emsg.go

```diff
@@ -60,6 +63,14 @@ func DecodeEmsgSR(hdr BoxHeader, startPos uint64, sr bits.SliceReader) (Box, err
 	} else {
 		return nil, fmt.Errorf("Unknown version for emsg")
 	}
+
+	// Initial box size does not include the message data length
+	remainingBytes := hdr.Size - b.Size()
```

Comment:
@tobbee comparing to your proposal: I use calculated size of box without the `MessageData` field to get the remaining bytes szie.

Your proposal was:
```
currPos := sr.GetPos()
nrBytesRead := currPos - initPos + 8
remainingSize := int(hdr.Size) - int(nrBytesRead)
```

Any objections for using b.Size() method here ?

Reply:
Well, Size() is supposed to give the true size of the box, while you here use it in an intermediate step. Looks fragile to me, so I prefer not using it.

Reply:
Changed accordingly. 👍🏻